src/imitation/algorithms/adversarial/common.py

```diff
@@ -434,10 +435,15 @@ def train(
         Args:
             total_timesteps: An upper bound on the number of transitions to sample
                 from the environment during training.
-            callback: A function called at the end of every round which takes in a
-                single argument, the round number. Round numbers are in
-                `range(total_timesteps // self.gen_train_timesteps)`.
+            callback: List of stable_baslines3 callback to be passed to the policy
+                learning function.
         """
+        if callback is not None:
+            if self.gen_callback is None:
+                self.gen_callback = callback
+            else:
+                self.gen_callback = callback + [self.gen_callback]
```

Comment:
Can someone abuse the API by calling `train()` multiple times? If so, the value of `self.gen_callback` would contain nested list, which is not correct. Generally, the value of gen_callback is currently `Optional[BaseCallback]` and we shouldn't  change the type to a list at runtime.

Perhaps it would be better to add an optional callback argument to `train_gen()`, merge callbacks there, and avoid the stateful change here?

Reply:
Also, can the `learn_kwargs` argument from `train_gen()` be removed, as discussed in the original issue #607 ?